### PR TITLE
[DROOLS-1186] Reproducer for 4th incarnation of NPE

### PR DESCRIPTION
**Pull request by yurloc**
I guess this shouldn't be merged without a fix for the failing test.

### CODE REVIEW

File: drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java [hunk 1]
```diff
@@ -8498,6 +8499,94 @@ public void testDeletedRightTupleInChangedBucket() {
     }
 
     @Test
+    public void test() {
+        // DROOLS-1186
```
Comment (yurloc) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java:8503]: Any suggestions for a good name?

File: drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java [hunk 2]
```diff
@@ -8498,6 +8499,94 @@ public void testDeletedRightTupleInChangedBucket() {
...
+        kieSession.fireAllRules();
+
+        setEmployee(kieSession, shift8, x);
+        setEmployee(kieSession, shift4, x);
+        // 6 must be after 7, otherwise order doesn't matter
+        setEmployee(kieSession, shift7, x);
+        setEmployee(kieSession, shift6, x);
+        // ----result-----
+        // 1 2 3 4 5 6 7 8
+        // x x o x x x x x
```
Comment (yurloc) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java:8579]: I've included these comments in the hope they will help to find the root cause more quickly. Feel free to remove them when merging.

File: drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java (hunk 1 above)
Comment (triceo) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java:8503]: Can we stop growing the `Misc2Test` class? I hope I don't live long enough to see `Misc3Test`.

Comment (ge0ffrey) [inline drools-compiler/src/test/java/org/drools/compiler/integrationtests/Misc2Test.java:8503]: worst case scenario, name it `drools1186()`.

+1 to split up MiscTest into ExistsMiscTest, NotMiscTest, AccumulateMiscTest, etc (or a better split up)

**Comment by yurloc** [comment]
Superseded by #826.

**Pull request closed without merging**